cypress/e2e/audio-player/audio-player.spec.ts

```diff
@@ -0,0 +1,563 @@
+        ".mx_SeekBar, " +
+        ".mx_JumpToBottomButton, .mx_MessageTimestamp, .mx_RoomView_myReadMarker { visibility: hidden !important; }";
+
+    const visitRoom = () => {
+        cy.visit("/#/room/" + roomId);
+        cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Group);
+
+        // Wait until configuration is finished
+        cy.contains(
+            ".mx_RoomView_body .mx_GenericEventListSummary[data-layout='group'] .mx_GenericEventListSummary_summary",
+            "created and configured the room.",
+        ).should("exist");
+    };
+
+    const uploadFile = (file: string) => {
+        // Upload a file from the message composer
+        cy.get(".mx_MessageComposer_actions input[type='file']").selectFile(file, { force: true });
+
+        cy.get(".mx_Dialog").within(() => {
+            // Click "Upload" button
+            cy.get("[data-testid='dialog-primary-button']").should("have.text", "Upload").click();
+        });
+
+        // Wait until the file is sent
+        cy.get(".mx_RoomView_statusArea_expanded").should("not.exist");
+        cy.get(".mx_EventTile.mx_EventTile_last .mx_EventTile_receiptSent").should("exist");
+    };
+
+    const checkPlayerFilenameLong = () => {
+        // Detect the audio file
+        cy.get(".mx_EventTile_mediaLine .mx_MAudioBody").within(() => {
+            // Assert that the audio player is rendered
+            cy.get(".mx_AudioPlayer_container").within(() => {
+                // Assert that media information is rendered
+                cy.get(".mx_AudioPlayer_mediaInfo").within(() => {
+                    cy.get(".mx_AudioPlayer_mediaName").should("have.text", "1sec-long-name-audio-file.ogg");
+                    cy.contains(".mx_AudioPlayer_byline", "00:01").should("exist");
+                    cy.contains(".mx_AudioPlayer_byline", "(3.56 KB)").should("exist"); // actual size
+                });
+
+                // Assert that the play button is rendered
+                cy.get("[data-testid='play-pause-button'][aria-label='Play']").should("exist");
+            });
+        });
+    };
+
+    // Take snapshots in modern and bubble layout, outputting log for reference/debugging
+    // Player on IRC layout should have the same layout as group layout
+    const takeSnapshots = (wrapper: string, detail: string) => {
+        // Check the status of the seek bar
+        // TODO: check if visible - currently visibility check on a narrow timeline causes an error
+        cy.get(`${wrapper}.mx_AudioPlayer_seek input[type='range']`).should("exist");
+
+        // Assert that the pause button is not rendered
+        cy.get(`${wrapper}[data-testid='play-pause-button'][aria-label='Pause']`).should("not.exist");
+
+        // Assert that the play button is rendered
+        cy.get(`${wrapper}[data-testid='play-pause-button'][aria-label='Play']`).should("exist");
```

Comment:
These assertions seem to be unrelated to the business of taking a snapshot, which is the purpose of this function. Do you think they could/should they be moved into `checkPlayerFilenameLong` ?

Reply:
It looks like I am too careful not to create a flaky snapshot test (as I am not sure what causes the flakiness)... I'll move those assertions.

Reply:
After some checks, it seems that it is crucial to ensure that the play button is visible before taking a snapshot. This was addressed with https://github.com/matrix-org/matrix-react-sdk/pull/10441/commits/f6ffc1a3801c9e766f03084dc70a5ecc9dc1dba1.